src/driver/virtualbox.rs

```diff
@@ -32,6 +34,67 @@ impl Introspectable for VBox {
         self.fdp.get_physical_memory_size()
     }
 
+    fn read_registers(&self, vcpu: u16) -> Result<Registers, Box<dyn Error>> {
+        let fdp_vcpu: u32 = vcpu as u32;
```

Comment:
Why this redefinition?

Reply:
Oh, never mind. You could loose the type of the variable though:
```suggestion
        let fdp_vcpu = vcpu as u32;
```